core/src/main/java/io/questdb/griffin/engine/functions/str/SplitPartFunctionFactory.java

```diff
@@ -0,0 +1,243 @@
+
+        int argCount = argFunctions.getDynamicArgs().size();
+        if (argCount == 3) {
+            return new SplitPartTernaryFunction(argFunctions, strArg, delimiterArg, indexArg);
+        } else if (argCount == 2) {
+            return new SplitPartBinaryFunction(argFunctions, strArg, delimiterArg, indexArg);
+        } else if (argCount == 1) {
+            return new SplitPartUnaryFunction(argFunctions, strArg, delimiterArg, indexArg);
+        } else if (argCount == 0) {
+            return new SplitPartFunction(argFunctions, strArg, delimiterArg, indexArg);
+        } else {
+            throw new RuntimeException("Unexpected number of args: " + argCount);
+        }
+    }
+
+    private static class SplitPartTernaryFunction extends SplitPartFunction implements TernaryFunction {
+        public SplitPartTernaryFunction(ArgFunctions argFunctions,
+                                        ArgFunctions.Arg<CharSequence> strArg,
+                                        ArgFunctions.Arg<CharSequence> delimiterArg,
+                                        ArgFunctions.Arg<Integer> indexArg) {
+            super(argFunctions, strArg, delimiterArg, indexArg);
+        }
+
+        @Override
+        public Function getLeft() {
+            return argFunctions.getDynamicFunc(0);
+        }
+
+        @Override
+        public Function getCenter() {
+            return argFunctions.getDynamicFunc(1);
+        }
+
+        @Override
+        public Function getRight() {
+            return argFunctions.getDynamicFunc(2);
+        }
+    }
+
+    private static class SplitPartBinaryFunction extends SplitPartFunction implements BinaryFunction {
+        public SplitPartBinaryFunction(ArgFunctions argFunctions,
+                                       ArgFunctions.Arg<CharSequence> strArg,
+                                       ArgFunctions.Arg<CharSequence> delimiterArg,
+                                       ArgFunctions.Arg<Integer> indexArg) {
+            super(argFunctions, strArg, delimiterArg, indexArg);
+        }
+
+        @Override
+        public Function getLeft() {
+            return argFunctions.getDynamicFunc(0);
+        }
+
+        @Override
+        public Function getRight() {
+            return argFunctions.getDynamicFunc(1);
+        }
+    }
+
+    private static class SplitPartUnaryFunction extends SplitPartFunction implements UnaryFunction {
+        public SplitPartUnaryFunction(ArgFunctions argFunctions,
+                                       ArgFunctions.Arg<CharSequence> strArg,
+                                       ArgFunctions.Arg<CharSequence> delimiterArg,
+                                       ArgFunctions.Arg<Integer> indexArg) {
+            super(argFunctions, strArg, delimiterArg, indexArg);
+        }
+
+        @Override
+        public Function getArg() {
+            return argFunctions.getDynamicFunc(0);
+        }
+    }
+
+    private static class SplitPartFunction extends StrFunction {
```

Comment:
Same consideration on redundant child classes as in `SplitPartCharFunctionFactory`.